### PR TITLE
PLAT-25591 Set up granular profiling for Jep C code

### PR DESCRIPTION
**Pull request by ihleonard-c3**

### CODE REVIEW

File: c3scripts/c3-jep-debug [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+set -e
+
+echo "This does not work properly"
+exit 1
+#echo "NOTE: If you see the following message:\n"
+#echo "Error calling thread_get_state for GP registers for thread 0x1603\n\t\t\n\t\t\n\t\twarning: Mach error at '../../gdb/i386-darwin-nat.c:132' in function 'virtual void i386_darwin_nat_target::fetch_registers(struct regcache *, int)': (os/kern) invalid argument (0x4)\n\t\t--Type <RET> for more, q to quit, c to continue without paging--"
+#echo "Just enter 'c' to continue."
+#echo ""
+#echo "'sudo' is required to run GDB or dtrace; see https://github.com/cs01/gdbgui/issues/55 for more details. Please enter your password."
+#sudo -v
```
Comment (TomTJarosz) [inline c3scripts/c3-jep-debug:11]: Delete? Idrc but I'll point it out

File: c3scripts/c3-jep-test [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
+TEST=${2:-C3JepInterface_Test}
 
-conda activate jep_dev && python setup.py install
-
-cd $C3_SERVER_ROOT
+# Assume ./c3-conda install has been run
+#CONDA_BASE=$(conda info --base)
+#source $CONDA_BASE/etc/profile.d/conda.sh
+#
+#conda activate jep_dev
```
Comment (TomTJarosz) [inline c3scripts/c3-jep-test:10]: Delete? Idrc but I'll point it out

File: c3-jep [hunk 3]
```diff
@@ -3,29 +3,28 @@
...
+echo "usage: `basename $0` <command> [ PARAMS ]"
 echo ""
 echo "  commands:"
 echo ""
 for cmd in "${cmds[@]}" ; do
   echo "${cmd}"
 done
 echo ""
+echo "  Note: The default test for test, debug, and profile is C3JepInterface_Test"
+echo "  To run a different test, simply pass it as an argument after the command"
```
[the remark this answers is not in the document]
Comment (ihleonard-c3) [reply c3-jep:27]: Yeah, let me add a bit

File: c3scripts/c3-jep-debug (hunk 1 above)
Comment (ihleonard-c3) [reply c3scripts/c3-jep-debug:11]: Just the `sudo -v`? Or the whole section. As we discussed earlier, I was having some issues with debugging and couldn't figure out whether it was my setup, this script, some combination...

File: README.md [hunk 4]
```diff
@@ -12,11 +12,12 @@ will create the environment if it does not exist.
 To run the c3server tests that exercise the `(C3)JepInterface` features, run `./c3-jep test`.
```
[the remark this answers is not in the document]
Comment (ihleonard-c3) [reply README.md:13]: Yup

File: c3scripts/c3-jep-debug (hunk 1 above)
Comment (TomTJarosz) [reply c3scripts/c3-jep-debug:11]: I was referring to the whole section, but I only meant to point out that the code is commented out; if you feel the commented out code is no longer relevant, remove it. If not, then leaving it is fine.

Comment (ihleonard-c3) [reply c3scripts/c3-jep-debug:11]: If you're ok with it, I'd like to leave it for now until we figure out why my debugging wasn't/isn't working

**Pull request merged**